### PR TITLE
[DC-3607] Create cleaning rule to remove notes from CDR

### PR DESCRIPTION
**Pull request by MichaelSchmidt82**
Notes submitted by sites and RDR must be suppressed since they are not being released soon.
This cleaning rule replaces note_text with `NO_TEXT` and note_title with `NO_TITLE` in the combined stage.

This test checks all four cases:

1. An unacceptable value in note_title
2. An inappropriate value in note_text
3. Both note_title and note_text contain wrong values.
4. Both note_title and note_text contain acceptable values.
5. Both note_title and note_text are empty strings `""`

Original Issues: DC-3607

### CODE REVIEW

**Review by ksdkalluri — CHANGES_REQUESTED** [review]
Looks good @MichaelSchmidt82 I suggested a slight modification to your query and suggested 2 more test scenarios for the integration test.

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/replace_freetext_notes_test.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                note_title,
+                note_text,
+                encoding_concept_id,
+                language_concept_id
+        )
+        VALUES
+              (1, 10, DATE('2023-01-01'), TIMESTAMP('2023-01-01 00:00:00'), 0, 0, 'bad note_title entry', 'NO_TEXT', 0, 0),
+              (2, 20, DATE('2023-01-01'), TIMESTAMP('2023-01-01 00:00:00'), 0, 0, 'NO_TITLE', 'bad note_text entry', 0, 0),
+              (3, 30, DATE('2023-01-01'), TIMESTAMP('2023-01-01 00:00:00'), 0, 0, 'bad note_title entry', 'bad note_text entry', 0, 0),
+              (4, 40, DATE('2023-01-01'), TIMESTAMP('2023-01-01 00:00:00'), 0, 0, 'NO_TITLE', 'NO_TEXT', 0, 0)
```
[the remark this answers is not in the document]
Comment (MichaelSchmidt82) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/replace_freetext_notes_test.py:86]: Gotcha

File: data_steward/cdr_cleaner/cleaning_rules/replace_freetext_notes.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+from utils import pipeline_logging
+from common import JINJA_ENV, NOTE
+
+LOGGER = logging.getLogger(__name__)
+JIRA_ISSUE_NUMBERS = ['DC3607']
+
+FREE_TEXT_UPDATE_QUERY = JINJA_ENV.from_string("""
+UPDATE `{{project_id}}.{{dataset_id}}.note`
+  SET
+    note_title = CASE WHEN NOT REGEXP_CONTAINS('note_title', '(?i)NO_TITLE') THEN 'NO_TITLE'
```
[the remark this answers is not in the document]
Comment (MichaelSchmidt82) [reply data_steward/cdr_cleaner/cleaning_rules/replace_freetext_notes.py:23]: thanks for that.  I felt it was costly

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/replace_freetext_notes_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichaelSchmidt82) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/replace_freetext_notes_test.py]: archiving `NULL` is an error as those fields are required.

**Review by ksdkalluri — APPROVED** [review]
Good to merge @MichaelSchmidt82 !

**Pull request merged**